src/OpenTelemetry.Contrib.Instrumentation.AWS/OpenTelemetry.Contrib.Instrumentation.AWS.csproj

```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard2.0</TargetFrameworks>
+    <Description>AWS client instrumentation for OpenTelemetry .NET</Description>
+	<MinVerTagPrefix>Instrumentation.AWS-</MinVerTagPrefix>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\OpenTelemetry.Contrib.Extensions.AWSXRay\OpenTelemetry.Contrib.Extensions.AWSXRay.csproj" />
```

Comment:
Why does this instrumentation depend on the Extensions.Awsxray? which part would need it?

Reply:
The AWS client instrumentation needs `AWSXRayPropagator` so that the trace header to the downstream AWS services is propagated in X-Ray format.
https://github.com/srprash/opentelemetry-dotnet-contrib/blob/aws_sdk_instrumentation/src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs#L149

Reply:
But that's a job for a propagator, not for instrumentation.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/MicroserviceExample/Utils/Messaging/MessageSender.cs#L78 You probably want something like this. Setting the global propagator to be XRay one - its not instrumentation's task.

Reply:
If a downstream service expects the trace context, the instrumentation does need to call the `Inject` method on the propagator right?
Taking ref to the http instrumentation: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L89

Reply:
> https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/MicroserviceExample/Utils/Messaging/MessageSender.cs#L78 You probably want something like this. Setting the global propagator to be XRay one - its not instrumentation's task.

I'm doing something similar. we don't want to set the XRay propagator as global because the user application could be using some other propagator, that's why just within the AWS instrumentation, I'm using the XRay propagator.

Reply:
If X-RayPropagator is **only** required within this instrumentation, you can make propagator as part of this instrumentation, not even make it public. But i am not sure why you want to only limit it inside instrumentation. Propagators support composite ones, so if it is not meeting the needs, we need to understand more.

Reply:
So let me try to explain the use case of XRay propagator. The XRay propagator comes into play for 2 cases:
1. Consider 2 services A and B. Service A is instrumented with OpenTelemetry and Service B is instrumented with X-Ray. In order to effectively propagate trace context from A to B via an http call, service A **must** use XRayPropagator because service B can only understand X-Ray format trace header on the http request. Here service A explicitly needs to set XRay propagator to be used to inject the trace context into the http request headers.
2. Consider 2 services A and B both instrumented with OpenTelemetry. In addition to this Service A also invokes AWS Lambda through an instrumented AWS client. In this case, the request to AWS Lambda **must** use XRayPropagator to propagate the X-Ray format trace. But service A need not set XRay propagator as the global propagator because a TextMapPropagator would work just fine between A and B. 

Hope this helps in undersatnding the role of AWSXRayPropagator as per my understanding. If I got something wrong or missed some concept, please shed some light on it.

Reply:
Similar explanation for point 2 in Java instrumentation for AWS SDK Java clients: https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/aws-sdk/aws-sdk-2.2/library#trace-propagation

Reply:
If using AWS client instrumentation, it's required to configure `AWSXRayPropagator` globally in user's `startup.cs` file or elsewhere, so it's reasonable to call `Propagators.DefaultTextMapPropagator.Inject` here, instead of referencing `OpenTelemetry.Contrib.Extensions.AWSXRay.csproj` again.

Reply:
And we have to specify that requirement in user's guide if they would like to trace AWS calls.

Reply:
@lupengamzn Since we know the hard requirement of using `AWSXRayPropagator` for AWS SDK instrumentation, I believe we should handle the complexity ourselves on customer's behalf rather than asking the customer to do the work. 
For that reason the reference to the Extensions.AWSXRay project is required so that the customer automatically gets everything they need by simply pulling in the Instrumentation.AWS package.

Reply:
Customer will need `AWSXRayPropagator` as well for tracing http call, in that case, they have to configure the propagator globally anyway.

Reply:
Also, since separating the AWS client instrumentation from the `Extension.XRay` project mainly targets to specify each part's function, so I believe it is better to let `Extension.XRay` do the propagation configuration work and ask customer to set `AWSXRayPropagator` globally as a basic requirement as long as they would like to get application traced by X-Ray as `AWSXRayPropagator` is required at multiple places. And the customer's behavior is also more uniform no matter they have AWS client instrumentation in their application or not.

Reply:
> Consider 2 services A and B. Service A is instrumented with OpenTelemetry and Service B is instrumented with X-Ray. In order to effectively propagate trace context from A to B via an http call, service A must use XRayPropagator because service B can only understand X-Ray format trace header on the http request. Here service A explicitly needs to set XRay propagator to be used to inject the trace context into the http request headers.

serviceA can use CompositePropagator which has XRayPropagator inside it (along with Baggage/TraceContext ones)